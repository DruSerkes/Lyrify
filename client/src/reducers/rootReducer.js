import songReducer from './songReducer';
import userReducer from './userReducer';
import favoritesReducer from './favoritesReducer';
import { combineReducers } from 'redux';

const rootReducer = combineReducers({ song: songReducer, user: userReducer, favorites: favoritesReducer });

export default rootReducer;
