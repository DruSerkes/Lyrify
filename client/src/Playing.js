import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import Box from '@material-ui/core/Box';
import { getNowPlaying } from './reducers/actions';
import Lyrics from './Lyrics';
import Paper from '@material-ui/core/Paper';

const Playing = () => {
	const dispatch = useDispatch();
	const songData = useSelector((state) => state.song.data);
	const [ fetching, setFetching ] = useState(false);

	const handleGetNowPlaying = () => {
		setFetching(true);
	};

	useEffect(
		() => {
			// if (songData === {} && !fetching) {
			// 	setFetching(true);
			// }
			if (fetching) {
				dispatch(getNowPlaying());
			}
			return () => {
				setFetching(false);
			};
		},
		[ dispatch, fetching ]
	);

	return (
		<Box padding={2} margin={2}>
			<Typography variant="h1">Now Playing</Typography>
			<Button onClick={handleGetNowPlaying}>Get Now Playing</Button>
			{songData === {} ? (
				<Paper>
					<Typography variant="h5">No lyrics found</Typography>
				</Paper>
			) : (
				<Lyrics artist={songData.artist} song={songData.song} lyrics={songData.lyrics} />
			)}
		</Box>
	);
};
