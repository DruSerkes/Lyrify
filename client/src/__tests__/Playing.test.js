import React from 'react';
import { render } from '@testing-library/react';
import Playing from '../Playing';
import { MemoryRouter } from 'react-router-dom';
import rootReducer from '../reducers/rootReducer';
import thunk from 'redux-thunk';
import { applyMiddleware, createStore } from 'redux';
import { Provider } from 'react-redux';
import { composeWithDevTools } from 'redux-devtools-extension';

const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)));
const songData = {
	id     : 1,
	song   : 'test song',
	artist : 'the testers'
};
it('renders without crashing', () => {
	render(
		<Provider store={store}>
			<MemoryRouter>
				<Playing songData={songData} userId={1} />
			</MemoryRouter>
		</Provider>
	);
});

it('matches snapshot', () => {
	const { asFragment } = render(
		<Provider store={store}>
			<MemoryRouter>
				<Playing songData={songData} userId={1} />
			</MemoryRouter>
		</Provider>
	);
	expect(asFragment()).toMatchSnapshot();
});
