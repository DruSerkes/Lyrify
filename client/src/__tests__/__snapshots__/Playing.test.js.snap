// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-5 Playing"
  >
    <div
      class="MuiBox-root MuiBox-root-6 Playing-Header"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1"
      >
        Now Playing
      </h1>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Get Current Song
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <section
      class="MuiBox-root MuiBox-root-7 Lyrics"
    >
      <div
        class="MuiPaper-root Lyrics MuiPaper-elevation5 MuiPaper-rounded"
      >
        <section
          class="Lyrics-Content"
        >
          <div
            class="Lyrics-Header"
          >
            <div
              class="Lyrics-Header-Top"
            >
              <h4
                class="MuiTypography-root MuiTypography-h4"
              >
                "test song"
              </h4>
              <h5
                class="MuiTypography-root MuiTypography-h5"
              >
                By: the testers
              </h5>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-middle"
          />
          <div
            class="MuiBox-root MuiBox-root-8"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Cannot find lyrics for this song
            </p>
          </div>
        </section>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
